modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -305,6 +344,34 @@ CrackFrontDefinition::initialSetup()
 void
 CrackFrontDefinition::initialize()
 {
+  // Below is copied from CrackFrontDefinition::initialSetup()
+  // The purpose is to prepare for J-integral calculations using updated crack front
+  if (_use_mesh_cutter)
+  {
+    if (_crack_front_points_provider != nullptr)
+    {
+      _crack_front_points =
+          _crack_front_points_provider->getCrackFrontPoints(_num_points_from_provider);
+      if (_use_mesh_cutter)
+        _crack_plane_normals =
+            _crack_front_points_provider->getCrackPlaneNormals(_num_points_from_provider);
+    }
+
+    updateCrackFrontGeometry();
+
+    std::size_t num_crack_front_points = getNumCrackFrontPoints();
+    if (_q_function_type == "GEOMETRY")
+    {
+      if (!_treat_as_2d)
+        if (num_crack_front_points < 1)
+          mooseError("num_crack_front_points is not > 0");
+      for (std::size_t i = 0; i < num_crack_front_points; ++i)
+      {
+        bool is_point_on_intersecting_boundary = isPointWithIndexOnIntersectingBoundary(i);
+        _is_point_on_intersecting_boundary.push_back(is_point_on_intersecting_boundary);
+      }
+    }
+  }
```

Comment:
I further simplified the code here so it won't be a duplication (about 15 lines in `initialize()` vs. 46 lines in `initialsetup()`).  I thought about defining a new method/function to include the duplicated lines, but these lines are not together in `initialsetup()` so it may not be necessarily better.